File: Lista_Tarefa/app/src/main/java/com/example/lista_tarefa/activity/MainActivity.java
package com.example.lista_tarefa.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.DividerItemDecoration;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.AdapterView;
import android.widget.LinearLayout;

import com.example.lista_tarefa.R;
import com.example.lista_tarefa.adapter.TarefaAdapter;
import com.example.lista_tarefa.helper.RecyclerItemClickListener;
import com.example.lista_tarefa.model.Tarefa;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private RecyclerView recyclerView;
    private TarefaAdapter tarefaAdapter;
    private List<Tarefa> listaTarefas = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        recyclerView = findViewById(R.id.recyclerView);


        recyclerView.addOnItemTouchListener(new RecyclerItemClickListener(getApplicationContext(), recyclerView, new RecyclerItemClickListener.OnItemClickListener() {
            @Override
            public void onItemClick(View view, int position) {
                Log.i("clique", "onItemClick");
            }

            @Override
            public void onLongItemClick(View view, int position) {
                Log.i("clique", "onLongItemClick");
            }

            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {

            }
        }));

        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(getApplicationContext(), AddTarefa.class);
                startActivity(intent);
            }
        });
    }

    public void carregarListatarefas(){

        Tarefa tarefa1 = new Tarefa();
        tarefa1.setNomeTarefa("Comprar pizza");
        listaTarefas.add(tarefa1);

        Tarefa tarefa2 = new Tarefa();
        tarefa1.setNomeTarefa("Comprar pizza 2");
        listaTarefas.add(tarefa2);

        tarefaAdapter = new TarefaAdapter(listaTarefas);

        RecyclerView.LayoutManager layoutManager =  new LinearLayoutManager(getApplicationContext());
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setHasFixedSize(true);
        recyclerView.addItemDecoration(new DividerItemDecoration(getApplicationContext(), LinearLayout.VERTICAL));
        recyclerView.setAdapter(tarefaAdapter);
    }

    @Override
    protected void onStart() {
        carregarListatarefas();
        super.onStart();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}

File: Prova2/app/src/main/java/com/example/prova/MainActivity.kt
package com.example.prova

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        passar_dados.setOnClickListener {

            val intent = Intent(applicationContext, ResultadoActivity::class.java)

            val precoEtanol = editTextEtanol.text.toString()
            val precoGasolina = editTextGasolina.text.toString()

            intent.putExtra("Gasolina", precoEtanol)
            intent.putExtra("Etanol", precoGasolina)

            val validaCampo = validaCampo(precoEtanol, precoGasolina)

            if(validaCampo == true){
                startActivity(intent)
            }else{
                textViewError.setText("Preencha todos os campos campos primeiro")
            }
        }
    }

    fun validaCampo(etanol: String, gasolina: String) : Boolean{
        var camposValidos:Boolean = true

        if(etanol == null || etanol.equals("") && gasolina == null || gasolina.equals("")){

            camposValidos = false

        }
        return camposValidos
    }
}